flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/strategies/SpecificInputTypeStrategies.java

```diff
@@ -46,6 +55,15 @@
     public static final InputTypeStrategy CURRENT_WATERMARK =
             new CurrentWatermarkInputTypeStrategy();
 
+    /** Input strategy for {@link BuiltInFunctionDefinitions#JSON_OBJECT}. */
+    public static final InputTypeStrategy JSON_OBJECT =
+            compositeSequence()
+                    .argument(symbol(JsonOnNull.class))
+                    .finishWithVarying(
+                            repeatingSequence(
+                                    and(logical(LogicalTypeFamily.CHARACTER_STRING), LITERAL),
```

Comment:
Perhaps can you add a comment here being explicit about how the signature of JSON_OBJECT function works and its inference? e.g.: 

> JSON_OBJECT takes as input a vararg composed by a sequence of couples of string keys and any value

Or something like that...

Reply:
I've extended the JavaDoc for it to describe it in plain words.